raft/src/main/java/org/apache/kafka/raft/RaftManager.java

```diff
@@ -0,0 +1,41 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.kafka.raft;
+
+import org.apache.kafka.common.protocol.ApiMessage;
+import org.apache.kafka.common.requests.RequestContext;
+import org.apache.kafka.common.requests.RequestHeader;
+import org.apache.kafka.server.common.serialization.RecordSerde;
+
+import java.util.concurrent.CompletableFuture;
+
+public interface RaftManager<T> {
+
+    CompletableFuture<ApiMessage> handleRequest(
+        RequestContext context,
+        RequestHeader header,
+        ApiMessage request,
+        long createdTimeMs
+    );
+
+    RaftClient<T> client();
+
+    ReplicatedLog replicatedLog();
```

Comment:
Please add comments to remind developers that those methods return the same closable objects. Therefore, callers should NOT close them